### PR TITLE
THREESCALE-1791 - 404 Application cannot be deleted on Keycloak

### PR DESCRIPTION
**Pull request by macejmic**
Fixes [THREESCALE-1791](https://issues.jboss.org/browse/THREESCALE-1791)

### CODE REVIEW

File: app/services/integration/keycloak_service.rb [hunk 1]
```diff
@@ -38,7 +38,7 @@ def self.call(entry)
     attr_reader :tenant, :client_id, :scope
 
     def initialize(entry)
-      @client_id = entry.data&.dig('client_id')
+      @client_id = (entry.data || entry.previous_data)&.dig('client_id')
```
Comment (Martouta) [inline app/services/integration/keycloak_service.rb]: We can fix the nil check 😄 
```suggestion
      @client_id = (entry.data || entry.previous_data || {}).dig('client_id')
```

Comment (Martouta) [reply app/services/integration/keycloak_service.rb]: And maybe we should even add a method in `entry` for this 😄 
image

Maybe something like:
```ruby
class Entry
  def last_known_client_id
    (data || previous_data || {}).dig('client_id')
  end
end
```
And here: `entry.last_known_client_id`

Comment (mikz) [reply app/services/integration/keycloak_service.rb]: Entry is generic for any type as zync can sync more than just Applications. So Entry should not really know about `client_id` (but surely can know about previous data).

Comment (Martouta) [reply app/services/integration/keycloak_service.rb]: Ok, then maybe:
```ruby
class Entry
  def last_known_data
    (data || previous_data || {})
  end
end
```
and then here:
`entry.last_known_data.dig('client_id')`

**Pull request merged**